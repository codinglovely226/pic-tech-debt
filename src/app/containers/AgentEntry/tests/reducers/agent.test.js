/* eslint-disable */

import agentRecordReducer from '../../reducers/agent';

import {
  addAgent,
  addAgentCentreAgent,
  removeAgentCentreAgent,
  updateAgentCategory,
  updateAgentCentreAgent,
  updateAgentDuration,
  updateAgentEUPCS,
  updateAgentIntendedUseGiven,
  updateAgentInterlocutorAgentName,
  updateAgentIsPatientsOwnMedication,
  updateAgentUseType,
  updateQuickViewAgentName,
} from '../../actions/case';

describe('agentRecordReducer', () => {
  it('returns the initial state', () => {
    expect(agentRecordReducer(undefined, {})).toMatchSnapshot();
  });
  it('add agent', () => {
    expect(agentRecordReducer({'id': 1}, addAgent(2, 'Test Agent', '1', 'mg', false, '2018-04-14 14:25:00', '1', 'hour', false, 'Test Exposure Type', 'Ingested', '123456'))).toMatchSnapshot();
  });
  it('add agent centre agent', () => {
    expect(agentRecordReducer({'id': 1}, addAgentCentreAgent(1, 1, 'Test Agent', '', '1', 'mg'))).toMatchSnapshot();
  });
  it('remove agent centre agent', () => {
    expect(agentRecordReducer({'id': 1}, removeAgentCentreAgent(1, 1, 1, 'Test Agent'))).toMatchSnapshot();
  });
  it('update agent category', () => {
    expect(agentRecordReducer({'id': 1}, updateAgentCategory(1, 1, 'Old Category', 'New Category'))).toMatchSnapshot();
  });
  it('update agent centre agent', () => {
    expect(agentRecordReducer({'id': 1}, updateAgentCentreAgent(1, 1, 1, 'Test Agent', 'Test URL', '1', 'mg'))).toMatchSnapshot();
  });
  it('update agent eupcs', () => {
    expect(agentRecordReducer({'id': 1}, updateAgentEUPCS(1, 1, 1, '1', '2', 'Old EUPCS', 'New EUPCS'))).toMatchSnapshot();
  });
  it('update agent intended use given', () => {
    expect(agentRecordReducer({'id': 1}, updateAgentIntendedUseGiven(1, 1, 'Old Agent Name', 'New Agent Name'))).toMatchSnapshot();
  });
  it('update agent interlocutor agent name', () => {
    expect(agentRecordReducer({'id': 1}, updateAgentInterlocutorAgentName(1, 1, 'Old Agent Name', 'New Agent Name'))).toMatchSnapshot();
  });
  it('update agent is patients own medication', () => {
    expect(agentRecordReducer({'id': 1}, updateAgentIsPatientsOwnMedication(1, 1, '', "Patient's own medication"))).toMatchSnapshot();
  });
  it('update agent use type', () => {
    expect(agentRecordReducer({'id': 1, 'type': ['Old Type']}, updateAgentUseType(1, 1, ['Old Use Type'], ['New Use Type']))).toMatchSnapshot();
  });
  it('update quick view agent name', () => {
    expect(agentRecordReducer({'id': 1, 'name': 'Old Name'}, updateQuickViewAgentName([{}]))).toMatchSnapshot();
  });
});
