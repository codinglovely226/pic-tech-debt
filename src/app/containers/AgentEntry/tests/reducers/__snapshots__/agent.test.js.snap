exports[`agentRecordReducer add agent 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "1",
    "units": "mg",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "1",
    "durationUnits": "hour",
    "durationUnknown": false,
    "route": "Ingested",
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "2018-04-14 14:25:00",
    "type": "Test Exposure Type",
  },
  "id": "123456",
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "Test Agent",
  "type": Array [],
}
`;

exports[`agentRecordReducer add agent centre agent 1`] = `
Object {
  "category": "",
  "centreAgents": Array [
    Object {
      "name": "Test Agent",
      "quantity": "1",
      "units": "mg",
      "url": "",
    },
  ],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "",
  "type": Array [],
}
`;

exports[`agentRecordReducer remove agent centre agent 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "",
  "type": Array [],
}
`;

exports[`agentRecordReducer returns the initial state 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": "",
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "",
  "type": Array [],
}
`;

exports[`agentRecordReducer update agent category 1`] = `
Object {
  "category": "New Category",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "",
  "type": Array [],
}
`;

exports[`agentRecordReducer update agent centre agent 1`] = `
Object {
  "category": "",
  "centreAgents": Array [
    Object {
      "name": "Test Agent",
      "quantity": "1",
      "units": "mg",
      "url": "Test URL",
    },
  ],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "",
  "type": Array [],
}
`;

exports[`agentRecordReducer update agent eupcs 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "1",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "",
  "type": Array [],
}
`;

exports[`agentRecordReducer update agent intended use given 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "New Agent Name",
  "isPatientsOwnMedication": false,
  "name": "",
  "type": Array [],
}
`;

exports[`agentRecordReducer update agent interlocutor agent name 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "Old Agent Name",
  "type": Array [],
}
`;

exports[`agentRecordReducer update agent is patients own medication 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": "Patient\'s own medication",
  "name": "",
  "type": Array [],
}
`;

exports[`agentRecordReducer update agent use type 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": "",
  "type": Array [
    "New Use Type",
  ],
}
`;

exports[`agentRecordReducer update quick view agent name 1`] = `
Object {
  "category": "",
  "centreAgents": Array [],
  "dosage": Object {
    "quantity": "",
    "units": "",
    "unknown": false,
  },
  "eupcs": "",
  "exposure": Object {
    "duration": "",
    "durationUnits": "",
    "durationUnknown": false,
    "route": Array [],
    "timeSince": "",
    "timeSinceUnits": "",
    "timestamp": "",
    "type": "",
  },
  "id": 1,
  "intendedUseGiven": "",
  "isPatientsOwnMedication": false,
  "name": Array [
    Object {},
  ],
  "type": Array [],
}
`;
